### PR TITLE
Generate changelog from makefile

### PR DESCRIPTION
**Pull request by johanbrandhorst**
In the same vein as the grpc-gateway, simply generate the changelog. Reduces the amount of work involved in making a new release and general maintenance, at the cost of a neatly curated changelog. Opinions welcome.

Steps for making a new release (with this):

1. Create a new branch, generate CHANGELOG.md with
   ```bash
   $ NEW_RELEASE=<new_release> CHANGELOG_GITHUB_TOKEN=<token> make changelog
   ```
2. Merge CHANGELOG.md changes.
3. Tag new release at the new commit.

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -15,3 +15,18 @@ test: vet
 	./scripts/test_all.sh
 
 .PHONY: all test
+
+changelog:
```
Comment (yashrsharma44) [inline makefile:19]: `docker` is supposed to be an optional dependency? Else to use this package, we might need to install docker, which might be a big ask 😛

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,51 +1,219 @@
...
+- retry doesnt seems to work  [\#154](https://github.com/grpc-ecosystem/go-grpc-middleware/issues/154)
+- grpc\_retry:not work as a calloption in my situation [\#151](https://github.com/grpc-ecosystem/go-grpc-middleware/issues/151)
+- Downstream interceptors are not re-executed on retry [\#110](https://github.com/grpc-ecosystem/go-grpc-middleware/issues/110)
+- Can we get semver releases? [\#40](https://github.com/grpc-ecosystem/go-grpc-middleware/issues/40)
+
+**Merged pull requests:**
+
+- Quick Fix of \#228 [\#229](https://github.com/grpc-ecosystem/go-grpc-middleware/pull/229) ([domgreen](https://github.com/domgreen))
+- remove last net/context reference will help fix \#175 [\#227](https://github.com/grpc-ecosystem/go-grpc-middleware/pull/227) ([domgreen](https://github.com/domgreen))
+- Move from dep to modules [\#226](https://github.com/grpc-ecosystem/go-grpc-middleware/pull/226) ([domgreen](https://github.com/domgreen))
```
Comment (yashrsharma44) [inline CHANGELOG.md:92]: Can we have the old style arrangments of PRs? 😅 
Like - `- [#226](https://github.com/grpc-ecosystem/go-grpc-middleware/pull/226) Support for go modules.`

```suggestion
- [#226](https://github.com/grpc-ecosystem/go-grpc-middleware/pull/226) Support for go modules.
```

**Review by yashrsharma44 — COMMENTED** [review]
This would be a good addition for generating changelogs. +1 for this 💯

**Comment by bwplotka** [comment]
Nice, I proposed removing it, let me know if this makes sense: https://github.com/grpc-ecosystem/go-grpc-middleware/pull/542

Rationale: GH releases allows nice generation of CHANGELOG straight there.

**Comment by bwplotka** [comment]
Alternative was merged, closing.

**Pull request closed without merging**